ddsrouter_core/test/blackbox/ddsrouter_core/dds/InitialPeers/DDSTestInitialPeers.cpp

```diff
@@ -0,0 +1,481 @@
+                transport_protocol)
+        );
+    }
+
+    if (is_server(wan_kind))
+    {
+        listening_addresses.insert(
+            types::Address(
+                (ip_version == types::IpVersion::v4 ? "127.0.0.1" : "::1"),
+                11666 + (this_server_id_is_1 ? 1u : 0u),
+                ip_version,
+                transport_protocol)
+            );
+    }
+
+    if (tls)
+    {
+        return std::make_shared<configuration::InitialPeersParticipantConfiguration>(
+            types::ParticipantId("InitialPeersParticipant_" + std::to_string((this_server_id_is_1 ? 1 : 0))),
+            types::ParticipantKind::wan_initial_peers,
+            false,
+            domain,
+            listening_addresses,
+            connection_addresses,
+            tls_configuration(wan_kind));
+
+    }
+    else
+    {
+        return std::make_shared<configuration::InitialPeersParticipantConfiguration>(
+            types::ParticipantId("InitialPeersParticipant_" + std::to_string((this_server_id_is_1 ? 1 : 0))),
+            types::ParticipantKind::wan_initial_peers,
+            false,
+            domain,
+            listening_addresses,
+            connection_addresses,
+            types::security::TlsConfiguration());
+    }
+}
+
+/**
+ * @brief Create a simple configuration for a DDS Router
+ *
+ * Create a configuration with 1 topic
+ * Create 1 simple participants with domains \c domain
+ * Create 1 custom participant by the configuration in \c participant_configuration
+ *
+ * @return configuration::DDSRouterConfiguration
+ */
+configuration::DDSRouterConfiguration router_configuration(
+        std::shared_ptr<configuration::ParticipantConfiguration> participant_configuration,
+        types::DomainIdType domain)
+{
+    // One topic
+    std::set<std::shared_ptr<types::FilterTopic>> allowlist(
+                    {
+                        std::make_shared<types::WildcardTopic>(TOPIC_NAME, "HelloWorld"),
+                    });
+
+    std::set<std::shared_ptr<types::FilterTopic>> blocklist;   // empty
+
+    std::set<std::shared_ptr<types::RealTopic>> builtin_topics;   // empty
+
+    // Two participants, one custom and other simple. If server, simple will work in 0, if not in 1
+    std::set<std::shared_ptr<configuration::ParticipantConfiguration>> participants_configurations(
+                    {
+                        // custom
+                        participant_configuration,
+
+                        // simple
+                        std::make_shared<configuration::SimpleParticipantConfiguration>(
+                            types::ParticipantId("simple_participant_" + std::to_string(domain)),
+                            types::ParticipantKind(types::ParticipantKind::simple_rtps),
+                            false,
+                            types::DomainId(domain)
+                            ),
+                    }
+        );
+
+    configuration::DDSRouterConfiguration configuration;
```

Comment:
I actually meant to have as in the original `DDSTestWAN.cpp`. Now it is using the default (12). I guess it doesn't matter.